### PR TITLE
Feature/wf 566 c modal

### PR DESCRIPTION
**Pull request by claudio147**
## Pull request
For the modal I used this plugin https://github.com/javisperez/vuedals
There are 2 new components `c-modal` and `c-modal-header-default`, for the c-modal you can add via prop a custom header.
There are 2 sizes for the modal according to the design
 
### Ticket
https://winterhalter-fenner.atlassian.net/browse/WF-566
 
### Browser testing
- http://localhost:6060/styleguide#cmodal
- http://localhost:8080/styleguide/modals
 
### Checklist
- [x] Added JSDoc and styleguide demo
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Did run automated tests and linters
- [x] Did assign ticket
- [x] Double checked target branch
 
## Review/Test checklist
- [x] Did run automated tests and linters
- [x] Did review code and documentation
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Re-assign ticket to developer

### CODE REVIEW

**Comment by claudio147** [comment]
@patric-eberle Please check again

File: app/components/c-modal.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  };
+</script>
+```
+
+#### No header
+```vue
+<template>
+  <div>
+  <e-button @click="modalState = true">open modal</e-button>
+    <c-modal :open="modalState" noCloseIcon @close="close">
```
Comment (patric-eberle) [inline app/components/c-modal.md]: - [ ] Please use `snake-case` in html

File: app/components/c-modal.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  };
+</script>
+```
+
+#### No Padding
+```vue
+<template>
+  <div>
+  <e-button @click="modalState = true">open modal</e-button>
+    <c-modal :open="modalState" noPadding noCloseIcon @close="close">
```
Comment (patric-eberle) [inline app/components/c-modal.md]: - [ ] Please use `snake-case` in html

File: app/components/c-modal.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  };
+</script>
+```
+
+#### Not closable
+```vue
+<template>
+  <div>
+  <e-button @click="modalState = true">open modal</e-button>
+    <c-modal :open="modalState" noCloseIcon :closable="false" title="Example title" @close="close">
```
Comment (patric-eberle) [inline app/components/c-modal.md]: - [ ] Please use `snake-case` in html

File: app/setup/components.js [hunk 4]
```diff
@@ -7,6 +7,9 @@ import eInput from '@/components/e-input';
 import eCheckbox from '@/components/e-checkbox';
 import eLabel from '@/components/e-label';
 import eProgress from '@/components/e-progress';
+import eInfoLabel from '@/components/e-info-label';
+import cModal from '@/components/c-modal';
+import cModalHeaderDefault from '@/components/c-modal-header-default';
```
Comment (patric-eberle) [inline app/setup/components.js]: - [ ] This is already imported by `c-modal` should not be needed globally therefore.

File: app/components/e-info-label.vue [hunk 5]
```diff
@@ -0,0 +1,179 @@
+<template>
+  <div :class="b({ design })">
+    <div
+      :class="b('inner', { design, hasInfo })"
+      @click="onClick"
```
Comment (patric-eberle) [inline app/components/e-info-label.vue]: - [ ] Please don't bind events to non interactive elements

File: app/components/c-modal.vue [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+       */
+      title: {
+        type: String,
+        default: ''
+      },
+
+      /**
+       * Hides the close icon in the header
+       */
+      noCloseIcon: {
```
[the remark this answers is not in the document]
Comment (claudio147) [reply app/components/c-modal.vue]: @patric-eberle I've already a prop `closable` which says it's closable by "esc" or click outside the modal. `noCloseIcon` is a property for the header - or should we use the same property?

File: app/components/c-modal.vue [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+        min-width: 500px;
+      }
+
+      header {
+        border-bottom: 0;
+        min-height: 0;
+      }
+
+      &::after {
+        @include media(sm) {
```
[the remark this answers is not in the document]
Comment (claudio147) [reply app/components/c-modal.vue]: Looks like it's not needed anymore

File: app/setup/components.js (hunk 4 above)
Comment (claudio147) [reply app/setup/components.js]: @patric-eberle Without this import it's not working.

File: app/components/c-modal.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (patric-eberle) [reply app/components/c-modal.vue]: - [] as discussed `closable` should en-/disable all of them.

**Pull request merged**